Source/Plugin.BLE/Android/BroadcastReceivers/BondStatusBroadcastReceiver.cs

```diff
@@ -1,41 +1,59 @@
 using System;
 using Android.Bluetooth;
 using Android.Content;
-using Plugin.BLE.Abstractions;
+using Android.OS;
 using Plugin.BLE.Abstractions.EventArgs;
 using Plugin.BLE.Android;
+using Plugin.BLE.Extensions;
 
 namespace Plugin.BLE.BroadcastReceivers
 {
-    //[BroadcastReceiver]
     public class BondStatusBroadcastReceiver : BroadcastReceiver
     {
+	    private readonly Adapter _broadcastAdapter;
+
         public event EventHandler<DeviceBondStateChangedEventArgs> BondStateChanged;
 
+        public BondStatusBroadcastReceiver(Adapter adapter)
+        {
+            _broadcastAdapter = adapter;
+        }
+
         public override void OnReceive(Context context, Intent intent)
         {
-            var bondState = (Bond)intent.GetIntExtra(BluetoothDevice.ExtraBondState, (int)Bond.None);
-            //ToDo
-            var device = new Device(null, (BluetoothDevice)intent.GetParcelableExtra(BluetoothDevice.ExtraDevice), null);
-            Console.WriteLine(bondState.ToString());
+            if (BondStateChanged == null)
+            {
+	            return;
+            }
 
-            if (BondStateChanged == null) return;
+            var extraBondState = (Bond)intent.GetIntExtra(BluetoothDevice.ExtraBondState, (int)Bond.None);
 
-            switch (bondState)
+            BluetoothDevice bluetoothDevice;
+            
+            // As older versions of .NET (pre7) don't include the Tiramisu version code, handle it manually.
+#if NET6_0_OR_GREATER
+	        if (OperatingSystem.IsAndroidVersionAtLeast(33))
+#else
+            if ((int)Build.VERSION.SdkInt >= 33)
+#endif
             {
-                case Bond.None:
-                    BondStateChanged(this, new DeviceBondStateChangedEventArgs() { Device = device, State = DeviceBondState.NotBonded });
-                    break;
-
-                case Bond.Bonding:
-                    BondStateChanged(this, new DeviceBondStateChangedEventArgs() { Device = device, State = DeviceBondState.Bonding });
-                    break;
+#if NET7_0_OR_GREATER
+	            bluetoothDevice = (BluetoothDevice)intent.GetParcelableExtra(BluetoothDevice.ExtraDevice, Java.Lang.Class.FromType(typeof(BluetoothDevice)));
+#else
+	            bluetoothDevice = (BluetoothDevice)intent.GetParcelableExtra(BluetoothDevice.ExtraDevice);
+#endif
```

Comment:
Another stupid question here: Why is the additional condition on NET7.0 necessary here? The variant with two arguments is only available on Android 33+ AFAICS, and you already have a check on the Android version above.

It is probably necessary because we are not targeting Android 13 in the classic Xamarin context, right? So maybe we should really update the targeted Android version for Xamarin. I can look into that.

Reply:
> It is probably necessary because we are not targeting Android 13 in the classic Xamarin context, right? So maybe we should really update the targeted Android version for Xamarin. I can look into that.

see #740

Reply:
<img width="804" alt="image" src="https://github.com/dotnet-bluetooth-le/dotnet-bluetooth-le/assets/138866394/8c8e9783-9712-4623-9fb1-0d1a7043a165">  
Older apps don't know the new overload added in api 33 right?

When using #740 changes and then removing the version check it runs into the following problem:
```
/Users/tobi/1_Code/3rd-Party/BT-Dotnet/dotnet-bluetooth-le-jeffhaynes/Source/Plugin.BLE/Android/BroadcastReceivers/BondStatusBroadcastReceiver.cs(40,56): error CS1501: No overload for method 'GetParcelableExtra' takes 2 arguments [/Users/tobi/1_Code/3rd-Party/BT-Dotnet/dotnet-bluetooth-le-jeffhaynes/Source/Plugin.BLE/Plugin.BLE.csproj::TargetFramework=net6.0-android]
```
I'm not exactly sure if we can remove it even when using your changes. As the error may be a problem of my toolchain I pushed the MonoAndroid update with the removed check as it's own commit here. If the pipeline passes everything is fine, if not I'll revert it.

Reply:
Yeah.. The CI also crashed because of it:  
```
Error: /Users/runner/work/dotnet-bluetooth-le/dotnet-bluetooth-le/Source/Plugin.BLE/Android/BroadcastReceivers/BondStatusBroadcastReceiver.cs(40,56): 
error CS1501: No overload for method 'GetParcelableExtra' takes 2 arguments 
[/Users/runner/work/dotnet-bluetooth-le/dotnet-bluetooth-le/Source/Plugin.BLE/Plugin.BLE.csproj::TargetFramework=net6.0-android]
```

I reverted it. So the MonoAndroid update can be merged in its own branch.

Reply:
Ok, very nice. Thanks.

Reply:
I re-triggered the CI and it succeeded now. 🎉